src/cdk-experimental/menu/menu.ts

```diff
@@ -34,4 +43,33 @@ export class CdkMenu extends CdkMenuGroup {
 
   /** Event emitted when the menu is closed. */
   @Output() readonly closed: EventEmitter<void | 'click' | 'tab' | 'escape'> = new EventEmitter();
+
+  /** List of nested CdkMenuGroup elements */
+  @ContentChildren(CdkMenuGroup, {descendants: true})
+  private readonly _nestedGroups: QueryList<CdkMenuGroup>;
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    // If there are nested MenuGroup elements within this group
+    // we complete the change emitter in order to ensure that
+    // the this menu does not emit change events.
+    if (this._hasNestedGroups()) {
+      this.change.complete();
+    } else {
+      this._nestedGroups.changes.pipe(take(1)).subscribe(() => this.change.complete());
+    }
+  }
+
+  /** Return true if there are nested CdkMenuGroup elements within the Menu */
+  private _hasNestedGroups() {
+    return (
+      this._nestedGroups.length > 0 &&
+      // view engine has a bug where @ContentChildren will return the current element
+      // along with children if the selectors match - not just the children.
+      // Here we check to see if the first element is a CdkMenu in order to ensure
+      // that we return true iff there are child CdkMenuGroup elements.
```

Comment:
I would avoid putting comments _in_ the conditional expression, instead putting the comment above the `return`